php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -1898,23 +1918,31 @@ private function fieldJsonByteSize($field)
             if ($count !== 0) {
                 if (!GPBUtil::hasSpecialJsonMapping($this)) {
                     $size += 3;                              // size for "\"\":".
-                    $size += strlen($field->getJsonName());  // size for field name
+                    if ($options & Options::JSON_ENCODE_PRESERVE_PROTO_FIELDNAMES) {
+                        $size += strlen($field->getName());
+                    } else {
+                        $size += strlen($field->getJsonName());
+                    } // size for field name
                 }
                 $size += 2;  // size for "[]".
                 $size += $count - 1;                     // size for commas
                 $getter = $field->getGetter();
                 foreach ($values as $value) {
-                    $size += $this->fieldDataOnlyJsonByteSize($field, $value);
+                    $size += $this->fieldDataOnlyJsonByteSize($field, $value, $options);
                 }
             }
         } elseif ($this->existField($field) || GPBUtil::hasJsonValue($this)) {
             if (!GPBUtil::hasSpecialJsonMapping($this)) {
                 $size += 3;                              // size for "\"\":".
-                $size += strlen($field->getJsonName());  // size for field name
+                if ($options & Options::JSON_ENCODE_PRESERVE_PROTO_FIELDNAMES) {
+                    $size += strlen($field->getName());
+                } else {
+                    $size += strlen($field->getJsonName());
+                } // size for field name
```

Comment:
I might be missing something - why is this behavior changing? I thought we were only changing the behavior of allowing enums as ints, but this is doing something different. Is this resolving a separate bug?

Reply:
This is a separate bug. When I added a test for BC of current behaviour, I discovered that the existing boolean parameter (preserve proto fieldnames) works in the extension, and does not work in the native. This aligns the two.

Reply:
I've made a patch to do this. BUT, when using the extension there is no user-facing constant to represent the flags (`Internal\CodedOutputStream` does not exist in the extension). It should have the same FQN in both. If we want to avoid adding new files, perhaps exposing constants in a class like `Message` instead?